Change minSdkVersion to 4

when deploying my application to phone i get this gradle build error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugManifest'.
> Manifest merger failed : uses-sdk:minSdkVersion 16 cannot be smaller than version 23 declared in library [manager:react-native-fingerprint-android:unspecified] /Users/ant/github/pdr-manager/android/app/build/intermediates/exploded-aar/manager/react-native-fingerprint-android/unspecified/AndroidManifest.xml
  	Suggestion: use tools:overrideLibrary="io.jari.fingerprint" to force usage
```

simple enough, so i went ahead and added `<uses-sdk tools:overrideLibrary="io.jari.fingerprint" />` to my application's `AndroidManifest.xml` which got the build working again. 

something tells me that what i did (above) shouldn't be necessary, however. perhaps the minimum sdk version for this library should be lower?

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -15,7 +15,7 @@ android {
     buildToolsVersion "23.0.1"
 
     defaultConfig {
-        minSdkVersion 23
+        minSdkVersion 16
         targetSdkVersion 23
         versionCode 1
         versionName "1.0"

--- a/android/src/main/java/io/jari/fingerprint/FingerprintModule.java
+++ b/android/src/main/java/io/jari/fingerprint/FingerprintModule.java
@@ -2,10 +2,9 @@ package io.jari.fingerprint;
 
 import android.Manifest;
 import android.content.pm.PackageManager;
-import android.hardware.fingerprint.FingerprintManager;
-import android.os.CancellationSignal;
+import android.support.v4.hardware.fingerprint.FingerprintManagerCompat;
+import android.support.v4.os.CancellationSignal;
 import android.support.v4.app.ActivityCompat;
-import android.util.Log;
 
 import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.ReactApplicationContext;
@@ -19,36 +18,40 @@ import java.util.Map;
 
 @SuppressWarnings("MissingPermission")
 public class FingerprintModule extends ReactContextBaseJavaModule {
-    FingerprintManager fingerprintManager;
+    FingerprintManagerCompat fingerprintManager;
     ReactApplicationContext reactContext;
     CancellationSignal cancellationSignal;
     boolean isCancelled = false;
 
     final int FINGERPRINT_ACQUIRED_AUTH_FAILED = 999;
 
+    final int FINGERPRINT_ERROR_CANCELED = 5;
+
     public FingerprintModule(ReactApplicationContext reactContext) {
         super(reactContext);
 
         this.reactContext = reactContext;
-        this.fingerprintManager = reactContext.getSystemService(FingerprintManager.class);
+        this.fingerprintManager = FingerprintManagerCompat.from(reactContext);
     }
 
+    // Harcdoded from https://developer.android.com/reference/android/hardware/fingerprint/FingerprintManager.html#FINGERPRINT_ACQUIRED_GOOD
+    // So we don't depend on FingerprintManager directly
     @Override
     public Map<String, Object> getConstants() {
         final Map<String, Object> constants = new HashMap<>();
-        constants.put("FINGERPRINT_ACQUIRED_GOOD", FingerprintManager.FINGERPRINT_ACQUIRED_GOOD);
-        constants.put("FINGERPRINT_ACQUIRED_IMAGER_DIRTY", FingerprintManager.FINGERPRINT_ACQUIRED_IMAGER_DIRTY);
-        constants.put("FINGERPRINT_ACQUIRED_INSUFFICIENT", FingerprintManager.FINGERPRINT_ACQUIRED_INSUFFICIENT);
-        constants.put("FINGERPRINT_ACQUIRED_PARTIAL", FingerprintManager.FINGERPRINT_ACQUIRED_PARTIAL);
-        constants.put("FINGERPRINT_ACQUIRED_TOO_FAST", FingerprintManager.FINGERPRINT_ACQUIRED_TOO_FAST);
-        constants.put("FINGERPRINT_ACQUIRED_TOO_SLOW", FingerprintManager.FINGERPRINT_ACQUIRED_TOO_SLOW);
+        constants.put("FINGERPRINT_ACQUIRED_GOOD", 0);
+        constants.put("FINGERPRINT_ACQUIRED_IMAGER_DIRTY", 3);
+        constants.put("FINGERPRINT_ACQUIRED_INSUFFICIENT", 2);
+        constants.put("FINGERPRINT_ACQUIRED_PARTIAL", 1);
+        constants.put("FINGERPRINT_ACQUIRED_TOO_FAST", 5);
+        constants.put("FINGERPRINT_ACQUIRED_TOO_SLOW", 4);
         constants.put("FINGERPRINT_ACQUIRED_AUTH_FAILED", FINGERPRINT_ACQUIRED_AUTH_FAILED);
-        constants.put("FINGERPRINT_ERROR_CANCELED", FingerprintManager.FINGERPRINT_ERROR_CANCELED);
-        constants.put("FINGERPRINT_ERROR_HW_UNAVAILABLE", FingerprintManager.FINGERPRINT_ERROR_HW_UNAVAILABLE);
-        constants.put("FINGERPRINT_ERROR_LOCKOUT", FingerprintManager.FINGERPRINT_ERROR_LOCKOUT);
-        constants.put("FINGERPRINT_ERROR_NO_SPACE", FingerprintManager.FINGERPRINT_ERROR_NO_SPACE);
-        constants.put("FINGERPRINT_ERROR_TIMEOUT", FingerprintManager.FINGERPRINT_ERROR_TIMEOUT);
-        constants.put("FINGERPRINT_ERROR_UNABLE_TO_PROCESS", FingerprintManager.FINGERPRINT_ERROR_UNABLE_TO_PROCESS);
+        constants.put("FINGERPRINT_ERROR_CANCELED", FINGERPRINT_ERROR_CANCELED);
+        constants.put("FINGERPRINT_ERROR_HW_UNAVAILABLE", 1);
+        constants.put("FINGERPRINT_ERROR_LOCKOUT", 7);
+        constants.put("FINGERPRINT_ERROR_NO_SPACE", 4);
+        constants.put("FINGERPRINT_ERROR_TIMEOUT", 3);
+        constants.put("FINGERPRINT_ERROR_UNABLE_TO_PROCESS", 2);
         return constants;
     }
 
@@ -92,7 +95,7 @@ public class FingerprintModule extends ReactContextBaseJavaModule {
         try {
             isCancelled = false;
             cancellationSignal = new CancellationSignal();
-            fingerprintManager.authenticate(null, cancellationSignal, 0, new AuthenticationCallback(promise), null);
+            fingerprintManager.authenticate(null, 0, cancellationSignal, new AuthenticationCallback(promise), null);
         } catch (SecurityException secEx) {
             Exception exception = new Exception("App does not have the proper permissions. (did you add USE_FINGERPRINT to your manifest?)\nMore info see https://github.com/jariz/react-native-fingerprint-android");
             exception.initCause(secEx);
@@ -120,13 +123,15 @@ public class FingerprintModule extends ReactContextBaseJavaModule {
         }
     }
 
+
     @Override
     public String getName() {
         return "FingerprintAndroid";
     }
 
-    public class AuthenticationCallback extends FingerprintManager.AuthenticationCallback {
+    public class AuthenticationCallback extends FingerprintManagerCompat.AuthenticationCallback {
         Promise promise;
+
 
         public AuthenticationCallback(Promise promise) {
             this.promise = promise;
@@ -135,7 +140,7 @@ public class FingerprintModule extends ReactContextBaseJavaModule {
         @Override
         public void onAuthenticationError(int errorCode, CharSequence errString) {
             super.onAuthenticationError(errorCode, errString);
-            if(errorCode == FingerprintManager.FINGERPRINT_ERROR_CANCELED) {
+            if(errorCode == FINGERPRINT_ERROR_CANCELED) {
                 isCancelled = true;
             }
             if(promise == null) {
@@ -158,7 +163,7 @@ public class FingerprintModule extends ReactContextBaseJavaModule {
         }
 
         @Override
-        public void onAuthenticationSucceeded(FingerprintManager.AuthenticationResult result) {
+        public void onAuthenticationSucceeded(FingerprintManagerCompat.AuthenticationResult result) {
             super.onAuthenticationSucceeded(result);
             if(promise == null) {
                 throw new AssertionError("Tried to resolve the auth promise, but it was already resolved / rejected. This shouldn't happen.");

--- a/example/package.json
+++ b/example/package.json
@@ -8,9 +8,9 @@
   "dependencies": {
     "color": "^0.11.3",
     "react": "15.3.1",
-    "react-native": "^0.33.0",
+    "react-native": "0.33.0",
     "react-native-animatable": "^0.6.1",
     "react-native-fingerprint-android": "file:..",
-    "react-native-svg": "^4.3.1"
+    "react-native-svg": "4.3.1"
   }
 }

